File: java/recursions_backtrackings/Day2/Para_Func.java
package recursions_backtrackings.Day2;

import java.util.*;

public class Para_Func {

    /*
     * // ---------- Parameterused ----------
     * // sum of first n numbers...
     * public static void sumOfN(int i, int sum) {
     * if (i < 1) {
     * System.out.println("result = " + sum);
     * return;
     * }
     * sumOfN(i - 1, sum + i);
     * }
     * 
     * 
     * public static void main(String[] args) {
     * Scanner sc = new Scanner(System.in);
     * 
     * System.out.print("Enter n = ");
     * int n = sc.nextInt();
     * 
     * sumOfN(n, 0);
     * 
     * sc.close();
     * }
     * 
     */


    

    // ----------- Funtional -----------
    // sum of first n numbers...
    public static int sumOfN(int n) {
        if (n == 0)
            return 0;

        return n + sumOfN(n - 1);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter n = ");
        int n = sc.nextInt();

        int result = sumOfN(n);

        System.out.println("result = " + result);
    }

}

File: python/recursions_backtrackings/day2/para_Func.py
# ---------- Parameterised ------------
# sum of n numbers...
# def sumOfN(i, sumi):
#     if i < 1:
#         print("result = ", sumi)
#         return
#     sumOfN(i - 1, sumi + i)


# n = int(input("Enter n = "))
# sumOfN(n, 0)





# ---------- Functional ------------
# sum of n numbers...
def sumOfN(n):
    if n == 0:
        return 0
    return n + sumOfN(n - 1)



n = int(input("Enter n = "))
result = sumOfN(n)
print("result = ", result)